internal/controller/dnsrecord_controller.go

```diff
@@ -387,19 +408,18 @@ func (r *DNSRecordReconciler) applyChanges(ctx context.Context, dnsRecord *v1alp
 				return noRequeueDuration, err
 			}
 		}
-		dnsRecord.Status.ValidFor = validationRequeueTime.String()
```

Comment:
I'm tempted to revert all these changes in `applyChanges` since now that the updateStatus is making sure these values are set to a consistent value on error the changes here are not really required to fix the bug.

That all said, this does also fix an issue where it didn't return `validationRequeueTime`, and also makes more sense to leave these updates to after we actually apply changes and know what is happening.

Why would you set `ValidFor` if you are entering a validation loop? Does that not suggest it's valid, even though we haven't done the validation loop yet?

Reply:
I would think: If there has been an error, validFor should be 0 (or nil, or similar) because it is not valid.

Reply:
This isn't related to errors, ensuring it's set to 0 on error is dealt with in `calculateStatus`. 

What should `validFor` be set to between the time it applies changes and comes back round and validates that there are no new changes.

Reply:
that, actually, makes more sense than what we have now :thinking: and gives that bit of information for the future reconciliation [here](https://github.com/Kuadrant/dns-operator/pull/99/files#diff-adacb6a4589c65c2d853c0f59ca02d6b8fb50bb388b41de1d13e99b0d128cfe1R189) but in this case I'd prefer it to be 0 just like Phil suggested  even if we have no errors and just doing validation

Reply:
I think it should still be the couple of seconds. If we think of this as... how long I'm willing to leave this before rechecking that it is still like I expect it to be. Then this is the period of time until it goes back to the API right? Initially 5s or whatever, then 10, 20, 40. It's never really saying "it's valid" but instead more like "I believe it's valid".

Reply:
OK `validFor` as a name does not suggest what you are saying, but changing it here is not needed to resolve the bug so will add it back but inside the code where we expect it to validating not at the start before we do anything.

Reply:
added back, sets all the values to what is expected deepening on whether there are changes or not

Reply:
the `validFor` will be exponential once we validated the record, but before the validation occurs that should be 0 :thinking:  the "I'll poke API at this time" is the `queuedFor` field. We aren't using it at the moment but that could be changed. 

Right now it looks like this: 

- got a fresh record, applied changes.
-  Set `queuedAt` as the current reconcile time, `validFor` is set to the duration in which i will poke the API and `queuedFor` = `queuedAt`+`validFor` 
- got record for validation. if  `queuedAt`+`validFor` is before `time.Now()` 
- repeat step 2

We could ask the logic to check for premature arrival of the record based on the `queuedFor` instead of doing that math. So the `queuedAt` and `validFor` fields will be just for humans to read. This should let us to use the `validFor` in a way it is called - those values are validated and will be valid for that much since the `queuedAt`